network_speed.sh

```diff
@@ -1,74 +1,63 @@
 #!/bin/bash -
 
-CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# Get the directory of the current script
+CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
+# Source the helper functions from helpers.sh
 source "$CURRENT_DIR/helpers.sh"
 
-default_download_color="#[fg=green]"
-default_upload_color="#[fg=yellow]"
+# Retrieve tmux options for upload and download colors, set default if not found
+default_upload_color="$(get_tmux_option "@network_speed_upload_color" "#[fg=yellow]")"
+default_download_color="$(get_tmux_option "@network_speed_download_color" "#[fg=green]")"
 
-get_speed()
-{
-    # Consts
-    local THOUSAND=1024
-    local MILLION=1048576
-
-    local new=$1
-    local current=$2
-    local interval=$3
-    local vel=0
-
-    local format_string=$(get_tmux_option '@network_speed_format' "%05.2f")
-
-    if [ ! "$current" -eq "0" ]; then
-        vel=$(echo "$(( $new - $current )) $interval" | awk '{print ($1 / $2)}')
-    fi
-
-    local vel_kb=$(echo "$vel" $THOUSAND | awk '{print ($1 / $2)}')
-    local vel_mb=$(echo "$vel" $MILLION | awk '{print ($1 / $2)}')
-
-    result=$(printf "%05.2f > 99.99\n" $vel_kb | bc -l)
-    if [[ $result == 1 ]]; then
-        local vel_mb_f=$(printf $format_string $vel_mb)
-        printf "%s MB/s" $vel_mb_f
-    else
-        local vel_kb_f=$(printf $format_string $vel_kb)
-        printf "%s KB/s" $vel_kb_f
-    fi
-}
+# Retrieve tmux options for speed threshold and high speed color, set default if not found
+threshold_speed="$(get_tmux_option "@network_speed_threshold" "1.0")"
```

Comment:
I think the default value for this should be 0. If we default to 1.0, people may see red color for upload or download speed when they don't expect it to be. Let's not apply this threshold by default. If someone want this they can explicitly set the threshold. What do you think?

Reply:
Completely agree, hadn't really considered that yet but makes a lot of sense to me.